src/main/fc/config.c

```diff
@@ -190,34 +190,30 @@ static void validateAndFixConfig(void)
         featureDisable(FEATURE_GPS);
     }
 
-    if (systemConfig()->activeRateProfile >= CONTROL_RATE_PROFILE_COUNT) {
-        systemConfigMutable()->activeRateProfile = 0;
-    }
-    loadControlRateProfile();
-
-    if (systemConfig()->pidProfileIndex >= PID_PROFILE_COUNT) {
-        systemConfigMutable()->pidProfileIndex = 0;
-    }
-    loadPidProfile();
-
-    // Prevent invalid notch cutoff
-    if (currentPidProfile->dterm_notch_cutoff >= currentPidProfile->dterm_notch_hz) {
-        currentPidProfile->dterm_notch_hz = 0;
-    }
+    for (unsigned i = 0; i < PID_PROFILE_COUNT; i++) {
+        // Prevent invalid notch cutoff
+        if (pidProfilesMutable(i)->dterm_notch_cutoff >= pidProfilesMutable(i)->dterm_notch_hz) {
+            pidProfilesMutable(i)->dterm_notch_hz = 0;
+        }
 
 #ifdef USE_DYN_LPF
-    //PRevent invalid dynamic lowpass
-    if (currentPidProfile->dyn_lpf_dterm_min_hz > currentPidProfile->dyn_lpf_dterm_max_hz) {
-        currentPidProfile->dyn_lpf_dterm_min_hz = 0;
-    }
+        //Prevent invalid dynamic lowpass
+        if (pidProfilesMutable(i)->dyn_lpf_dterm_min_hz > pidProfilesMutable(i)->dyn_lpf_dterm_max_hz) {
+            pidProfilesMutable(i)->dyn_lpf_dterm_min_hz = 0;
+        }
+
+        if (pidProfilesMutable(i)->dyn_lpf_dterm_min_hz > 0) {
```

Comment:
Maybe we need to test too if min < max, if not the dyn is not enabled.

Reply:
Not sure what you mean, there is no min and max for the static filter.

Reply:
Sorry, you have reason. I was talking about the dyn filter. You verify just in the previous line if min < max and then you reset to zero it's value. So the next if only must check if not zero. You can dismiss my comments.